src/sentry/integrations/msteams/webhook.py

```diff
@@ -200,6 +202,55 @@ def handle_member_removed(self, request):
         integration.delete()
         return self.respond(status=204)
 
+    def make_action_data(self, identity, data):
+        action_data = {}
+        if data["actionType"] == "unresolve" or data["actionType"] == "unignore":
+            action_data = {"status": "unresolved"}
+            analytics.record("integrations.msteams.unresolve", actor_id=identity.user_id)
+        elif data["actionType"] == "resolve":
+            status = data["resolveInput"]
+            status_data = status.split(":", 1)
+            resolve_type = status_data[-1]
+
+            action_data = {"status": "resolved"}
+            if resolve_type == "inNextRelease":
+                action_data.update({"statusDetails": {"inNextRelease": True}})
+            elif resolve_type == "inCurrentRelease":
+                action_data.update({"statusDetails": {"inRelease": "latest"}})
+            analytics.record(
+                "integrations.msteams.resolve", actor_id=identity.user_id, resolve_type=resolve_type
```

Comment:
@KKaempen It seems odd to me that this function which does mapping from data to action would have the side effect of also calling `analytics.record`. I think it would be cleaner to have `make_action_data` be a pure function and let the caller function do the analytics.

Reply:
Agree ^

Reply:
That's fair. I put it in there because that's the only place where I actually differentiate between what the `actionType` is. I could move it out though.